backend/src/main/java/com/google/rolecall/services/SectionServices.java

```diff
@@ -178,19 +326,100 @@ public Section editSection(SectionInfo newSection) throws EntityNotFoundExceptio
   public void deleteSection(int id) throws EntityNotFoundException, InvalidParameterException {
     Section section = getSection(id);
 
+    Integer siblingId = section.getSiblingId();
+    if(siblingId != null) {
+      Optional<Position> test = positionRepo.findById(siblingId);
+      if(test.isEmpty()) {
+        // Revelation's internal Ballet/Position structure has already been deleted
+      } else {
+        // Revelation's internal Ballet/Position structure still exists
+        throw new InvalidParameterException(
+          "Cannot delete Ballet that is part of a Revelation");
+      }
+    }
+
     if(performanceSectionRepo.findFirstBySection(section).isPresent() || 
         castRepo.findFirstBySection(section).isPresent()) {
       throw new InvalidParameterException(
-          "Cannot delete Section if it has a Cast or is part of a Performance");
+          "Cannot delete Ballet if it has a Cast or is part of a Performance");
     }
 
     sectionRepo.deleteById(id);
   }
 
-  public SectionServices(SectionRepository sectionRepo, CastRepository castRepo,
-      SubCastRepository subCastRepo, PerformanceSectionRepository performanceSectionRepo,
+  private SiblingError updateSiblingPosition(Integer positionId,
+      String newName,     // null or "": keep existing name
+      Integer siblingId   // -1: keep existing. 0: remove reference
+  ) {
+    if(positionId != null) {
+      Optional<Position> queryPosition = positionRepo.findById(positionId);
+
+      if(queryPosition.isEmpty()) {
+        return SiblingError.SIBLING_MISSING;
+      } else {
+        try {
+          Position sibling = queryPosition.get();
+          Position updatedSibling = sibling.toBuilder()
+              .setId(positionId)
+              .setName(newName == null || newName.length() == 0 ? sibling.getName() : newName)
+              .setNotes(sibling.getNotes())
+              .setOrder(sibling.getOrder())
+              .setSiblingId(siblingId == -1 ? sibling.getSiblingId() : siblingId)
+              .setSize(sibling.getSiblingId())
+              .build();
+          positionRepo.save(updatedSibling);
+          System.out.printf("Saved Sibling Position %s at id %d SiblingId=%d\n",
+              newName == null || newName.length() == 0 ? sibling.getName() : newName,
+              positionId,
+              siblingId == -1 ? sibling.getSiblingId() : siblingId);
+        } catch (InvalidParameterException e) {
+          return SiblingError.OTHER_ERROR;
+        }
+      }
+    }
+    return SiblingError.OK;
+  }
+
+  private SiblingError updateSiblingSection(Integer sectionId,
+      String newName,     // null or "": keep existing name
+      Integer siblingId   // -1: keep existing. 0: remove reference
+  ) {
+    if(sectionId != null) {
+      Optional<Section> querySection = sectionRepo.findById(sectionId);
+
+      //System.out.printf("Section Update id=%d name=%s siblingIs=%d\n", sectionId, newName, siblingId);
```

Comment:
Please remove the comment or switch to log.

Reply:
Done.